gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala

```diff
@@ -170,7 +170,15 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
   // It's misleading with "Codegen" used. But we have to keep "WholeStageCodegen" prefixed to
   // make whole stage transformer clearly plotted in UI, like spark's whole stage codegen.
   // See buildSparkPlanGraphNode in SparkPlanGraph.scala of Spark.
-  override def nodeName: String = s"WholeStageCodegenTransformer ($transformStageId)"
+  override def nodeName: String = {
+    var fullName = s"WholeStageCodegenTransformer ($transformStageId)"
+    child.foreach(c => {
+      if (c.isInstanceOf[FileSourceScanExecTransformer]) {
```

Comment:
why it needs to add file source scan node?

Reply:
If a sql select from serval tables, it's hard to know which table does a stage read from by spark ui, as all stages have the same name WholeStageCodegenTransformer.

Reply:
If it is strongly needed, can we control the use of verbose name via a config? I think It should be disabled by default to make the UI brief for most scenarios. Right?